plugins/analysis/types/heartbeat/packet.go

```diff
@@ -0,0 +1,125 @@
+package heartbeat
+
+import "errors"
+
+var (
+	ErrMalformedHeartbeatPacket = errors.New("malformed heartbeat packet")
+	ErrTooManyNeighborsToReport = errors.New("too many neighbors to report in packet")
+)
+
+// A heartbeat packet
+type Packet struct {
+	OwnID       []byte
+	OutboundIDs [][]byte
+	InboundIDs  [][]byte
+}
+
+func Unmarshal(data []byte) (*Packet, error) {
```

Comment:
I know we do that a lot throughout the code, but an alternative would be to use protobuf to generate the marshal code (this is for example used in `autopeering`).
However, unless we are expecting that this changes a lot, it probably does not make sense to migrate this to protobuf.

Reply:
I think it is reasonable to assume that with new metrics, additional fields will be necessary in the heartbeat message, so let's keep it here for now.